### DISCUSSION TITLE
[Funnel Chart] Ability to Customize Denominator for Percentage in Funnel Visualization

### DISCUSSION THREAD

**Discussion by lianncheung** [apache/superset#18521] [Ideas]
Currently, the funnel report percentage is calculated using: 
The number at a given funnel step / 
Sum(everything in the funnel)

Example from [blog](https://preset.io/blog/2021-6-14-superset-nine-new-charts/): 
![image](https://user-images.githubusercontent.com/18264869/138730429-dc443c51-7c59-4967-93c2-5e75ab1191b8.png)
Here, the Discussed Pricing (900) gets divided by 11900 (sum of all events in the funnel) to get 7.56%. 

I would like the ability to customize the denominator here, to be divided by either:
1) 1700 (Created Test Account) - the **previous step in the funnel**, or 
2) Requested Demo (6100) - the **top level of the funnel**. 
These are two common ways to look at funnel charts that we are currently unable to see.

**Comment by junlincc** [comment]
Make sense! thanks for suggesting!

**Comment by vacom13** [comment]
Hi. I would like to contribute to this. However, I am a newbie so do you mind guiding me? @junlincc

**Comment by kizimenko** [comment]
![image](https://user-images.githubusercontent.com/4619041/147265113-3bcfd25f-99a0-43b0-931b-5256db9be90a.png)
Something like this in plotly funnel

**Reply by RahulVishK** [reply]
hii do u have any update on this ??

**Reply by saghatelian** [reply]
@kizimenko Did you, by any chance, get this type of funnel in Superset? Is anyone working on this? The percentage calculation of the default funnel visualization doesn't make any sense to me.

**Comment by nishds** [comment]
Hi, will this be part of the improvements for the next SS upgrade? Could someone link me the upgrade details please? Cheers

**Comment by crazyisjen** [comment]
Hi, will this be part of the improvements for the next SS upgrade? Could someone link me the upgrade details please? Cheers

**Comment by NishantSinghChandel** [comment]
Is there any update on this, or any other way around to display this calculation.

**Comment by Valenzione** [comment]
Hi, does anyone has any idea if this is coming soon? Current implementation doesn't make sense

**Reply by saghatelian** [reply]
I don't think so. I posted a reply 4+ months ago, and no one reacted to it.

**Comment by TopEngineer0417** [comment]
Good idea and Good implement for you!

**Comment by kainchow** [comment]
The latest version has added filters to the conversion rate calculation logic on this PR: https://github.com/apache/superset/pull/26290/commits/e54f29d60562049c24be99a6b3fdbcc05e54f589

**Comment by CharlesYe8848** [comment]
您好，你发的邮件我已收到。

**Comment by alwyngo** [comment]
The proportion of the top layer of the superset funnel chart is not 100%, but the percentage of each layer calculated based on the sum. how to change it？

**Comment by CharlesYe8848** [comment]
您好，你发的邮件我已收到。
